teuthology/task/ansible.py

```diff
@@ -250,12 +254,28 @@ def _build_args(self):
         return args
 
     def teardown(self):
+        if self.config.get("cleanup"):
+            self._cleanup()
         if self.generated_inventory:
             os.remove(self.inventory)
         if self.generated_playbook:
             os.remove(self.playbook_file.name)
         super(Ansible, self).teardown()
 
+    def _cleanup(self):
+        """
+        If the ``cleanup`` key exists in config the same playbook will be
+        ran again during the teardown step with the var ``teardown`` given with
```

Comment:
You're proposing I always call ``_cleanup`` in ``teardown``, but then noop in ``_cleanup`` if 'cleanup' isn't in config?

Sure, I can do that.